js/app.js

```diff
@@ -43,3 +48,42 @@ document.getElementById("starting-items").textContent = getRandomWithSeed(
   .map(items => `C${items.id} ${items.name}`)
   .join(", ");
 document.getElementById("end-room").textContent = endingRoom;
+document.getElementById("generate-seed").addEventListener("click", function() {
+  newUrl = `?seed=${generateANewSeed()}`;
+  redirect(newUrl);
+});
+document.getElementById("share").addEventListener("click", function() {
+  copyToClipboard();
+});
+
+function generateANewSeed() {
```

Comment:
I'd maybe rename this to just `generateRandomSeed`, as it better describes what it does (and it's rare to see names like "doAThing" with "A" in them).